sled-agent/src/bin/zone-bundle.rs

```diff
@@ -449,11 +449,7 @@ async fn main() -> anyhow::Result<()> {
                 .await
                 .context("failed to get zone bundle")?
                 .into_inner();
-            let mut f = tokio::fs::OpenOptions::new()
-                .read(true)
-                .write(true)
-                .create(true)
-                .open(&output)
+            let mut f = tokio::fs::File::create(&output)
```

Comment:
Changed this because Rust 1.77 produces a warning if `OpenOptions` is used but `truncate` isn't defined. In all of these cases we want exactly create + write + truncate = true, and `File::create` does this.